### ISSUE TITLE
Possible to use Digitalocean do-block-storage as backend?

### ISSUE THREAD

**Issue by saul-data** [longhorn/longhorn#3943]
I'd like to use the read write many mode of Longhorn inside a Digitalocean managed Kubernetes. The problem is when the nodes get replaced or restarted the folders or volumes dont necessarily persist. It is also not easy to attach a volume to a node outside of PVC with Digitalocean's managed K8s - so having the do-block-storage as a backend would be perfect.

I would like to use the do-block-storage as a backing storage with Longhorn RWX? Is that possible?

**Comment by PhanLe1010** [comment]
cc @c3y1huang Do you have any idea if DO block volume works with Longhorn when you investigate the issue https://github.com/longhorn/longhorn/issues/1800

**Comment by c3y1huang** [comment]
> cc @c3y1huang Do you have any idea if DO block volume works with Longhorn when you investigate the issue #1800

No idea. [#1800](https://github.com/longhorn/longhorn/issues/1800) is about EKS, GKE, AKS.

**Comment by PhanLe1010** [comment]
Will leave the ticket open to see if there is more interest in this topic

**Comment by Froxz** [comment]
@saul-data 

Did you find an answer/solution to this? I stuck with DO and mostly because of their upgrade policy they replace nodes, and Longhorn doesn't automatically move replicas, so maybe using Volumes will help...

**Comment by saul-data** [comment]
Yes 100% - I didn’t get an answer unfortunately. It will be hugely
beneficial.

I think there is a way to mount the volume on a location of the node where
longhorn picks it up - I think that’s worth experimenting with…

**Comment by Froxz** [comment]
@saul-data the problem is also DigitalOcean they will not re-attach any volume created outside k8s cluster.

What I'm doing now is decreasing `replicaReplenishmentWaitInterval` to 60 seconds

And create 2 pools in DO (based on DO they will upgrade cluster 1 pool after another). in this case, all volumes will be transferred to new nodes.) Anyway, I'm still testing this...

**Comment by saul-data** [comment]
Oh right! Yes that sounds messy. I agree with this concept of backing
longhorn or other Kubernetes storage mechanisms with volume storage by the
cloud providers.

**Comment by Froxz** [comment]
@saul-data Ideally it will be good if the Longhorn team will adopt DO csi for volumes: https://github.com/digitalocean/csi-digitalocean

**Comment by vasilyprokopov** [comment]
Any good solution or a workaround a few months later? Want to make sure before playing with two node pools.

**Comment by SergkeiM** [comment]
> Any good solution or a workaround a few months later? Want to make sure before playing with two node pools.

After Long playing with multiple providers (rook/ceph/longhorn/openebs)

I did the following:

DO K8S -> Rook with external ceph cluster (5 nodes 3 mng, 5 mons)

Rook on K8S just installs connect and creates volumes (RWM/RWO) on ceph cluster. and when nodes replaced storage is not affected.

**Comment by NedkoHristov** [comment]
Bumping the issue again.

**Comment by chokosabe** [comment]
This issue needs some love - from the above, is it the case that we can't use longhorn on digitalocean (without using node space). If so we should include that somewhere in the docs

**Comment by Hazmi35** [comment]
> > Any good solution or a workaround a few months later? Want to make sure before playing with two node pools.
> 
> After Long playing with multiple providers (rook/ceph/longhorn/openebs)
> 
> I did the following:
> 
> DO K8S -> Rook with external ceph cluster (5 nodes 3 mng, 5 mons)
> 
> Rook on K8S just installs connect and creates volumes (RWM/RWO) on ceph cluster. and when nodes replaced storage is not affected.

Hi, sorry for the bump. Where do you run the Ceph Cluster? Regular DO Droplets within the same VPC or what?

**Comment by SergkeiM** [comment]
> > > Any good solution or a workaround a few months later? Want to make sure before playing with two node pools.
> > 
> > 
> > After Long playing with multiple providers (rook/ceph/longhorn/openebs)
> > I did the following:
> > DO K8S -> Rook with external ceph cluster (5 nodes 3 mng, 5 mons)
> > Rook on K8S just installs connect and creates volumes (RWM/RWO) on ceph cluster. and when nodes replaced storage is not affected.
> 
> Hi, sorry for the bump. Where do you run the Ceph Cluster? Regular DO Droplets within the same VPC or what?

Yes I run regular DO droplets with attached Volume to each node, I run 3 nodes with attached volumes and 2 simple nodes (only 2 GB of Ram) for mons to have 5 total mons.
